api/v1beta1/grpc_disruption_test.go

```diff
@@ -0,0 +1,260 @@
+				{
+					TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/order",
+					ErrorToReturn:    "CANCELED",
+					OverrideToReturn: "{}",
+					QueryPercent:     99,
+				},
+			},
+		}
+
+		Expect(spec.Validate()).To(BeNil())
+	})
+
+	It("Query percent of 100 on two endpoints does validate", func() {
+		spec := v1beta1.GRPCDisruptionSpec{
+			Port: 50051,
+			Endpoints: []v1beta1.EndpointAlteration{
+				{
+					TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/order",
+					ErrorToReturn:    "CANCELED",
+					OverrideToReturn: "{}",
+					QueryPercent:     100,
+				},
+				{
+					TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/getCatalog",
+					ErrorToReturn:    "CANCELED",
+					OverrideToReturn: "{}",
+					QueryPercent:     100,
+				},
+			}}
+
+		Expect(spec.Validate()).To(BeNil())
+	})
+
+	It("Query percent which sums to 100 on two endpoints does validate", func() {
+		spec := v1beta1.GRPCDisruptionSpec{
+			Port: 50051,
+			Endpoints: []v1beta1.EndpointAlteration{
+				{
+					TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/order",
+					ErrorToReturn:    "CANCELED",
+					OverrideToReturn: "{}",
+					QueryPercent:     60,
+				},
+				{
+					TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/order",
+					ErrorToReturn:    "CANCELED",
+					OverrideToReturn: "{}",
+					QueryPercent:     40,
+				},
+			}}
+
+		Expect(spec.Validate()).To(BeNil())
+	})
+
+	It("Query percent which sums to less than 100 on two endpoints does validate", func() {
+		spec := v1beta1.GRPCDisruptionSpec{
+			Port: 50051,
+			Endpoints: []v1beta1.EndpointAlteration{
+				{
+					TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/order",
+					ErrorToReturn:    "CANCELED",
+					OverrideToReturn: "{}",
+					QueryPercent:     49,
+				},
+				{
+					TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/order",
+					ErrorToReturn:    "CANCELED",
+					OverrideToReturn: "{}",
+					QueryPercent:     49,
+				},
+			}}
+
+		Expect(spec.Validate()).To(BeNil())
+	})
+
+	It("Query percent which sums to more than 100 on two endpoints does validate", func() {
+		spec := v1beta1.GRPCDisruptionSpec{
+			Port: 50051,
+			Endpoints: []v1beta1.EndpointAlteration{
+				{
+					TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/order",
+					ErrorToReturn:    "CANCELED",
+					OverrideToReturn: "{}",
+					QueryPercent:     50,
+				},
+				{
+					TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/order",
+					ErrorToReturn:    "CANCELED",
+					OverrideToReturn: "{}",
+					QueryPercent:     51,
+				},
+			}}
+
+		Expect(spec.Validate()).ToNot(BeNil())
+	})
+
+	It("Bad error to return", func() {
+		spec := v1beta1.GRPCDisruptionSpec{
+			Port: 50051,
+			Endpoints: []v1beta1.EndpointAlteration{
+				{
+					TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/order",
+					ErrorToReturn:    "MEOW",
+					OverrideToReturn: "{}",
+					QueryPercent:     50,
+				},
+			},
+		}
+
+		Expect(spec.Validate()).ToNot(BeNil())
+	})
+
+	It("Good errors to return", func() {
```

Comment:
Should the query percents in this test be > 0?

Reply:
This should divide the errors evenly across 100 percent slots.

Reply:
(and therefore be a valid test)